plugins/modules/template_intent.py

```diff
@@ -2635,6 +2639,81 @@ def get_diff_deleted(self, config):
         self.status = "success"
         return self
 
+    def verify_diff_merged(self, config):
+        """
+        Validating the DNAC configuration with the playbook details
+        when state is merged (Create/Update).
+
+        Parameters:
+            config (dict) - Playbook details containing Global Pool,
+            Reserved Pool, and Network Management configuration.
+
+        Returns:
+            self
+        """
+
+        if config.get("configuration_templates") is not None:
+            template_available = self.get_have_project(config)
+            self.log(str(template_available))
+            if not template_available:
+                self.msg = "Configuration Template config is not applied to the DNAC."
+                self.status = "failed"
+                return self
+
+            self.get_have_template(config, template_available)
+            self.log("DNAC retrieved details: " + str(self.have_template.get("template")))
+            self.log("Playbook details: " + str(self.want.get("template_params")))
+            template_params = ["language", "name", "projectName", "softwareType",
+                               "softwareVariant", "templateContent"]
+            for item in template_params:
+                if self.have_template.get("template").get(item) != self.want.get("template_params").get(item):
+                    self.msg = " Configuration Template config is not applied to the DNAC."
+                    self.status = "failed"
+                    return self
+                self.result.get("response").update({"Validation": "Success"})
+
+        self.msg = "Successfully validated the Configuration Templates."
+        self.status = "success"
+        return self
+
+    def verify_diff_deleted(self, config):
+        """
+        Validating the DNAC configuration with the playbook details
+        when state is deleted (delete).
+
+        Parameters:
+            config (dict) - Playbook details containing Global Pool,
+            Reserved Pool, and Network Management configuration.
+
+        Returns:
+            self
+        """
+
+        if config.get("configuration_templates") is not None:
+            self.log("DNAC retrieved details: " + str(self.have))
+            self.log("Playbook details: " + str(self.want))
+            template_list = self.dnac_apply['exec'](
+                family="configuration_templates",
+                function="gets_the_templates_available",
+                params={"projectNames": config.get("projectName")},
+            )
+            if template_list and isinstance(template_list, list):
+                templateName = config.get("configuration_templates").get("template_name")
+                template_info = get_dict_result(template_list,
+                                                "name",
+                                                templateName)
+            if template_info:
```

Comment:
Can we move this if check inside the above if block. if template_list itself none, then also we check template_info in the current design which is not required.